File: Juego/Guerrero.java
package Juego;

public class Guerrero extends Personaje {
    private int daño;

    public Guerrero(String nombre, int puntosDeVida, int daño) {
        super(nombre, puntosDeVida);
        this.daño = daño;
    }

    @Override
    public void atacar(Personaje objetivo) {
        System.out.println(getNombre() + " ataca a " + objetivo.getNombre() + ".");
        objetivo.recibirDaño(daño);
    }
}

File: Juego/Guerrero.kt
package Juego

class Guerrero(nombre: String?, puntosDeVida: Int, private val daño: Int) :
    Personaje(nombre!!, puntosDeVida) {
    override fun atacar(objetivo: Personaje?) {
        println(nombre + " ataca a " + objetivo.nombre + ".")
        objetivo.recibirDaño(daño)
    }
}